client/state/account-recovery/reset/actions.js

```diff
@@ -8,20 +8,46 @@ import {
 	ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
 } from 'state/action-types';
 
-export function fetchResetOptions( userData ) {
-	return ( dispatch ) => {
-		dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST } );
+const fromApi = ( data ) => ( [
+	{
+		email: data.primary_email,
+		sms: data.primary_sms,
+	},
+	{
+		email: data.secondary_email,
+		sms: data.secondary_sms,
+	},
+] );
 
-		wpcom.undocumented().accountRecoveryReset( userData ).getResetOptions()
-			.then( options => dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE, options } ) )
-			.catch( error => dispatch( { type: ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR, error } ) );
-	};
-}
+export const fetchResetOptions = ( userData ) => ( dispatch ) => {
+	dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_REQUEST,
+	} );
 
-export function fetchResetOptionsByLogin( user ) {
-	return fetchResetOptions( { user } );
-}
+	return wpcom.req.get( {
+		body: userData,
+		apiNamespace: 'wpcom/v2',
+		path: '/account-recovery/lookup',
+	} ).then( data => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_RECEIVE,
+		options: fromApi( data ),
+	} ) )
+	.catch( error => dispatch( {
+		type: ACCOUNT_RECOVERY_RESET_OPTIONS_ERROR,
+		error,
+	} ) );
```

Comment:
Oops, that was a mistake. Thanks for catching them. They are both addressed by 597d3adf9fe278d9ca877ebbcb7b8405ba4ff1e3.